packages/core/src/abstract-dialect/query-generator-typescript.ts

```diff
@@ -443,16 +455,189 @@ export class AbstractQueryGeneratorTypeScript<Dialect extends AbstractDialect =
+      } else {
+        throw new TypeError(
+          'The include option for indexes must be an array or an sql expression.',
+        );
+      }
+    }
+
+    // The index will be incorrectly scoped if we don't specify the schema name,
+    // which will cause it to error if another schema contains a table that uses an index with an identical name
+    const quotedIndexName =
+      table.schema && this.dialect.supports.addIndex.schemaQuoted
+        ? // 'quoteTable' isn't the best name: it quotes any identifier.
+          // in this case, the goal is to produce '"schema_name"."index_name"' to scope the index in this schema
+          this.quoteTable({
+            schema: table.schema,
+            tableName: options.name || generateIndexName(table, indexOptions),
+          })
+        : this.quoteIdentifier(options.name || generateIndexName(table, indexOptions));
+
+    return joinSQLFragments([
+      'CREATE',
+      indexOptions.unique ? 'UNIQUE' : '',
+      indexOptions.type ? `${indexOptions.type} INDEX` : 'INDEX',
+      indexOptions.concurrently ? 'CONCURRENTLY' : '',
+      indexOptions.ifNotExists ? 'IF NOT EXISTS' : '',
+      quotedIndexName,
+      indexOptions.using ? `USING ${indexOptions.using}` : '',
+      `ON ${this.quoteTable(tableOrModel)}`,
+      `(${columnSql.join(', ')})`,
+      indexOptions.parser ? `WITH PARSER ${indexOptions.parser}` : '',
+      indexOptions.include ? includeSql : '',
+      indexOptions.where ? this.whereQuery(indexOptions.where) : '',
+    ]);
   }
 
   removeIndexQuery(
-    _tableName: TableOrModel,
-    _indexNameOrAttributes: string | string[],
-    _options?: RemoveIndexQueryOptions,
-  ): string {
-    throw new Error(`removeIndexQuery has not been implemented in ${this.dialect.name}.`);
+    tableOrModel: TableOrModel,
+    indexNameOrAttributes: string | string[],
+    options?: RemoveIndexQueryOptions,
+  ) {
+    if (options) {
+      rejectInvalidOptions(
+        'removeIndexQuery',
+        this.dialect,
+        REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS,
+        this.dialect.supports.removeIndex,
+        options,
+      );
+
+      if (options.cascade && options.concurrently) {
+        throw new Error(
+          `Cannot specify both concurrently and cascade options in removeIndexQuery.`,
+        );
+      }
+    }
+
+    let indexName;
+    const table = this.extractTableDetails(tableOrModel);
+    if (Array.isArray(indexNameOrAttributes)) {
+      indexName = generateIndexName(table, { fields: indexNameOrAttributes });
+    } else {
+      indexName = indexNameOrAttributes;
+    }
+
+    const quotedIndexName = table.schema
+      ? // 'quoteTable' isn't the best name: it quotes any identifier.
+        // in this case, the goal is to produce '"schema_name"."index_name"' to scope the index in this schema
```

Comment:
```suggestion
      ?
```
Probably you need to run prettier in this file after this change, so don't commit this suggestion directly

Reply:
Updated now